src/testRunner/unittests/tsserver/partialSemanticServer.ts

```diff
@@ -111,6 +111,46 @@ import { something } from "something";
             assert.isTrue(hasException);
         });
 
+        it("allows syntactic diagnostic commands", () => {
+            const file1: File = {
+                path: `${tscWatch.projectRoot}/a.ts`,
+                content: `if (a < (b + c) { }`
+            };
+            const configFile: File = {
+                path: `${tscWatch.projectRoot}/tsconfig.json`,
+                content: `{}`
+            };
+            const expectedErrorMessage = "')' expected.";
+
+            const host = createServerHost([file1, libFile, configFile]);
+            const session = createSession(host, {
+                serverMode: LanguageServiceMode.PartialSemantic,
+                useSingleInferredProject: true,
+                logger: createLoggerWithInMemoryLogs()
+            });
+
+            const service = session.getProjectService();
+            openFilesForSession([file1], session);
+            const request: protocol.SyntacticDiagnosticsSyncRequest = {
+                type: "request",
+                seq: 1,
+                command: protocol.CommandTypes.SyntacticDiagnosticsSync,
+                arguments: { file: file1.path }
+            };
+            const response = session.executeCommandSeq(request).response as protocol.SyntacticDiagnosticsSyncResponse["body"];
+            assert.isDefined(response);
+            assert.equal(response!.length, 1);
+            assert.equal((response![0] as protocol.Diagnostic).text, expectedErrorMessage);

```

Comment:
Technically we can remove these since they should be baselined in the server log. Thoughts?

Reply:
Personally, I think it's better to just use the baseline because updating this would be tedious.  However, I agree that it's hard to figure out which parts of the baseline a particular test cares about, so I find it helpful to document my expectations in the test so that someone bulk updating baselines has some chance of figuring out if they broke something.

Reply:
Looking back at the code, I think I'll keep it in because I have to check the language service directly anyway - not just the server.